enhancements/pipeline-tooling.md

```diff
@@ -0,0 +1,267 @@
+
+### Download the contents of an index
+
+As an Operator Registry maintainer, Pipeline admin, or QE, I want to download or inspect the contents of a given index image.
+
+#### Design Proposal
+
+Add a command to `opm index` and `opm alpha bundle` called `extract` which extracts:
+
+* the manifest and metadata folders for a given bundle
+
+    **Example CLI:**
+    
+    `opm alpha bundle extract quay.io/my/bundle:v1 --out path/to/my/dir`
+
+* a list of all the bundle images that are in an index
+    
+    **Example CLI 1:**
+    
+    `opm index extract quay.io/my/index --out path/to/my/dir --images-only`
+    
+    **Example CLI 2:** prints to stdout
+    
+    `opm alpha bundle extract quay.io/my/bundle:v1 --image-only`
+
+
+* the manifest and metadata folders for all bundles in a given index
+
+    **Example CLI:**
+    
+    `opm index extract quay.io/my/index --out path/to/my/dir`
+
+
+### Delete latest operator version
+
+As an Operator Registry maintainer, I want to have a way to remove the latest version of an operator (from an index)?
+
+#### Design Proposal
+
+Add a new flag to `opm registry rm` and `opm index rm` commands called `--latest` which:
+
+* takes an index reference and package name, and removes the latest bundle version if present.
+    
+    **Example CLI:**
+    `opm index rm --latest --from quay.io/my/index -o "operator1,operator2" -c=docker`
+
+* takes a registry database and package names, and removes the latest bundle version if present.
+
+    **Example CLI:**
+    `opm registry rm --latest -d=path/to/my.db -o="operator1,operator2"`
+    
+### Merging Indexes
+
+As an Operator Registry Maintainer I want to merge two indexes into one index.
+
+#### Design Proposal
+
+Initially we will only support disjoint indexes (i.e. indexes with no common packages) so that we don't have to deal with compatible versions and insertion order (which we currently cannot do implicitly). Once we change our update graph to rely solely on semver we could support merging intersecting indexes.
+
+We need to add a command to `opm index` called `merge` which takes two index images and outputs either a new index image or generates a directory and dockerfile that can be used to build an image.
+
+**Example 1:**
+
+`opm index merge quay.io/my/index1 quay.io/my/index2 -t quay.io/my/superindex -c=docker`
+
+**Example 2:**
+
+`opm index merge quay.io/my/index1 quay.io/my/index2 --generate -c=podman`
+
+### Get related images from an index
+
+As a pipeline admin, I want to know all the related images of a particular index so that I can know to update the index image when CVE fixes affect the related images of a bundle in my index.
+
+#### Design Proposal
+
+Add a flag to `opm index extract` called `related-images` that takes an index image and returns a list of related images from the database.
+
+**Example:**
+
+`opm index extract quay.io/my/index --related-images -c=docker`
```

Comment:
Should this be a flag and not its own command?

Reply:
How about a top-level `export` command that handles both indexes and bundles based on the content of the image?